src/index.js

```diff
@@ -0,0 +1,111 @@
+import { registerPlugin } from '@wordpress/plugins';
+import { PluginSidebar, PluginSidebarMoreMenuItem } from '@wordpress/edit-site';
+import { blockDefault } from '@wordpress/icons';
+import { Fragment, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { Button, MenuGroup, MenuItem, PanelBody, TextControl } from '@wordpress/components';
+import { store as noticesStore } from '@wordpress/notices';
+import { useDispatch } from '@wordpress/data';
+
+const GlobalStylesProvider = wp.editSite.GlobalStylesProvider;
+
+const StyleVariations = () => {
+	const [ variationName, setVariationName ] = useState( '' );
+	const [ canReset, onReset ] = wp.editSite.useGlobalStylesReset();
```

Comment:
It's nice for demo purposes but I'm wondering from a UX perspective, does it make sense to reset the styles when the variation is created?

Reply:
That's what we do in the PHP side of the tool.

Reply:
Ideally when in this perspective we could activate the variation (which would leave all of the styles as-set instead of resetting to default).